### PR TITLE
Modified the existing WebUI WIFI list.

### PR DESCRIPTION
**Pull request by rizwan2000rm**
Made changes to the existing table. I opted for a custom design rather than using a table as it looks too generic. I tried something modern and fancy with the design. Let me know any feedback to improve further upon it.

Things added to codebase
- [x] Responsive Design with modern look featuring dynamic icons, cards, modals etc.
- [x] Sorting based on different keys.

### CODE REVIEW

File: frontend/components/WifiDetails.jsx [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+  const handleDialog = () => {
+    setOpen(true);
+  };
+
+  return (
+    <>
+      <div className={`${classes.row} ${classes.wrapper}`}>
+        <div className={classes.row}>
+          <WifiIcon SIGNAL={parseInt(wifi_network.SIGNAL, 10)} />
+          {/* <SignalWifi4BarIcon color="primary" /> */}
```
Comment (rroohhh) [inline frontend/components/WifiDetails.jsx]: unnecessary comment

File: frontend/routes/Wifi.jsx [hunk 2]
```diff
@@ -1,53 +1,99 @@
...
+    margin: '20px 0',
+    display: 'flex',
+    flexDirection: 'column',
+    gap: '20px',
+  },
+}));
+
+export function Component(props) {
+  const classes = useStyles();
+  const [wifi_networks, setWifi_networks] = useState();
```
Comment (rroohhh) [inline frontend/routes/Wifi.jsx]: please name the variables consisten, either `snake_case` or `camelCase` not a mix.

File: frontend/routes/Wifi.jsx [hunk 3]
```diff
@@ -1,53 +1,99 @@
...
-    return (
-      <div>
-        <Table data={this.state.wifi_networks} />
-      </div>
-    );
-  }
-}
+  const handleChange = event => {
+    setSortBy(event.target.value);
+    setWifi_networks(sortArrayOfObjects(wifi_networks, event.target.value));
```
Comment (rroohhh) [inline frontend/routes/Wifi.jsx]: This only sorts the array once, it needs to be sorted everytime new wifi data is received.

File: frontend/util/sort.js [hunk 4]
```diff
@@ -0,0 +1,13 @@
+export const sortArrayOfObjects = (wifi_networks, key) => {
```
Comment (rroohhh) [inline frontend/util/sort.js]: I am not sure this should be a general util function, as
1. It has special handling for the `Default` sorting
2. it always uses numeric sorting, which might be quite unexpected 
Either change the name, or put it into `Wifi.jsx`

File: wifi-output.txt [hunk 5]
```diff
@@ -0,0 +1,38 @@
+[
```
Comment (rroohhh) [inline wifi-output.txt]: This file is unused, please remove it.

**Review by rroohhh — CHANGES_REQUESTED** [review]
Looks great so far, please see the inline comments.

Making the default sort put `IN_USE` first is fine with me.

File: frontend/components/WifiDetails.jsx [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+          <WifiIcon SIGNAL={parseInt(wifiNetwork.SIGNAL, 10)} />
+          <Typography className={wifiNetwork['IN-USE'] === '*' ? classes.active : ''}>
+            {wifiNetwork.SSID}
+          </Typography>
+        </div>
+        <div className={classes.row}>
+          <div className={classes.details}>
+            <Typography>{wifiNetwork.BSSID}</Typography>
+            <Typography component="div">
+              <Chip className={classes.chip} label={wifiNetwork.SECURITY} color="primary" />
```
Comment (rroohhh) [inline frontend/components/WifiDetails.jsx]: I think this would look better if there was one `Chip` per Security (for example some networks support WPA1 an PA2)

File: frontend/components/WifiDetails.jsx [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+      <div className={`${classes.row} ${classes.wrapper}`}>
+        <div className={classes.row}>
+          <WifiIcon SIGNAL={parseInt(wifiNetwork.SIGNAL, 10)} />
+          <Typography className={wifiNetwork['IN-USE'] === '*' ? classes.active : ''}>
+            {wifiNetwork.SSID}
+          </Typography>
+        </div>
+        <div className={classes.row}>
+          <div className={classes.details}>
+            <Typography>{wifiNetwork.BSSID}</Typography>
```
Comment (rroohhh) [inline frontend/components/WifiDetails.jsx:69]: I think the `BSSID` column would look better, if they would all line up, even if the `Security` entry for the rows have different lengths.

File: package.json [hunk 8]
```diff
@@ -14,23 +14,24 @@
...
     "express": "~4.17.1",
     "socketio": "^1.0.0"
   },
   "devDependencies": {
-    "@material-ui/core": "^4.8.0",
+    "@material-ui/core": "^4.9.10",
     "@material-ui/icons": "^4.5.1",
     "@reduxjs/toolkit": "^1.1.0",
-    "@types/react": "^16.9.17",
+    "@types/react": "^16.8.6 || ^17.0.0",
```
Comment (rroohhh) [inline package.json]: Where do these version constraints come from? 
Can you explain me the advantage of accepting both `^16.8.6` and `^17.0.0`?

**Review by rroohhh — CHANGES_REQUESTED** [review]

File: package.json (hunk 8 above)
Comment (rizwan2000rm) [reply package.json]: I don't remember changing any packages. This might be a mistake or something. I will revert it to either one.

File: frontend/components/WifiDetails.jsx [hunk 9]
```diff
@@ -0,0 +1,89 @@
...
+            <div className={classes.container}>
+              <Typography component="div">
+                {wifiNetwork.SECURITY === 'WPA1 WPA2' ? (
```
Comment (rroohhh) [inline frontend/components/WifiDetails.jsx]: What about cases, where there are different security options, like `WPA2 802.1X`, or `WPA2 WPA3`?

**Review by rroohhh — CHANGES_REQUESTED** [review]

File: frontend/components/WifiDetails.jsx (hunk 9 above)
Comment (rizwan2000rm) [reply frontend/components/WifiDetails.jsx]: Oh I actually didn't even though about it. Maybe the condition should not be so specific like "WPA2 WPA1" instead I think I should split the string on a blank space. It should work right ?

Comment (rroohhh) [reply frontend/components/WifiDetails.jsx]: Sure

File: frontend/components/WifiDetails.jsx [hunk 10]
```diff
@@ -68,14 +69,19 @@ export const WifiDetails = ({ wifiNetwork }) => {
             <Typography>{wifiNetwork.BSSID}</Typography>
             <div className={classes.container}>
               <Typography component="div">
-                {wifiNetwork.SECURITY === 'WPA1 WPA2' ? (
+                <div className={classes.row}>
+                  {wifiNetwork.SECURITY.split(' ').map(security => (
+                    <Chip className={classes.chip} label={security} color="primary" />
+                  ))}
+                </div>
+                {/* {wifiNetwork.SECURITY === 'WPA1 WPA2' ? (
```
Comment (rroohhh) [inline frontend/components/WifiDetails.jsx]: unnecessary comment

**Review by rroohhh — CHANGES_REQUESTED** [review]

File: frontend/components/WifiDetails.jsx (hunk 10 above)
Comment (rizwan2000rm) [reply frontend/components/WifiDetails.jsx]: I'm sorry this slipped by. I will resolve it.

File: frontend/components/WifiDetails.jsx [hunk 11]
```diff
@@ -8,7 +8,7 @@ import InfoIcon from '@material-ui/icons/Info';
 
 import { makeStyles } from '@material-ui/core/styles';
 
-const useStyles = makeStyles(theme => ({
+const useStyles = makeStyles((theme) => ({
```
Comment (rroohhh) [inline frontend/components/WifiDetails.jsx]: Why did you add parentheses here? Please make sure your commits are clean under `yarn format`

**Review by rroohhh — CHANGES_REQUESTED** [review]

File: frontend/components/WifiDetails.jsx (hunk 11 above)
Comment (rizwan2000rm) [reply frontend/components/WifiDetails.jsx]: Oh that might be my prettier extension in vs code using local config rather than project configuration. I did this edit on other machine so that might have added those parentheses. Will resolve it soon.
